src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractObjectCreationExpressionAnalyzer.cs

```diff
@@ -24,166 +22,44 @@ internal abstract class AbstractObjectCreationExpressionAnalyzer<
         where TObjectCreationExpressionSyntax : TExpressionSyntax
         where TVariableDeclaratorSyntax : SyntaxNode
     {
-        protected SemanticModel _semanticModel;
-        protected ISyntaxFacts _syntaxFacts;
+        protected UpdateExpressionState<TExpressionSyntax, TStatementSyntax> State;
+
         protected TObjectCreationExpressionSyntax _objectCreationExpression;
         protected bool _analyzeForCollectionExpression;
-        protected CancellationToken _cancellationToken;
-
-        protected TStatementSyntax _containingStatement;
-        private SyntaxNodeOrToken _valuePattern;
-        private ISymbol _initializedSymbol;
 
-        protected AbstractObjectCreationExpressionAnalyzer()
-        {
-        }
+        protected ISyntaxFacts SyntaxFacts => this.State.SyntaxFacts;
+        protected SemanticModel SemanticModel => this.State.SemanticModel;
 
-        protected abstract bool ShouldAnalyze();
-        protected abstract bool TryAddMatches(ArrayBuilder<TMatch> matches);
+        protected abstract bool ShouldAnalyze(CancellationToken cancellationToken);
+        protected abstract bool TryAddMatches(ArrayBuilder<TMatch> matches, CancellationToken cancellationToken);
```

Comment:
i don't like capturing cancellation tokens, so i made these be passed explicitly.